balancer/rls/balancer.go

```diff
@@ -297,13 +298,17 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 	if resizeCache {
 		// If the new config changes reduces the size of the data cache, we
 		// might have to evict entries to get the cache size down to the newly
-		// specified size.
+		// specified size. If we do evict an entry with valid backoff timer,
+		// the new picker needs to be sent to the channel to re-process any
+		// RPCs queued as a result of this backoff timer.
 		//
 		// And we cannot do this operation above (where we compute the
 		// `resizeCache` boolean) because `cacheMu` needs to be grabbed before
 		// `stateMu` if we are to hold both locks at the same time.
 		b.cacheMu.Lock()
-		b.dataCache.resize(newCfg.cacheSizeBytes)
+		if b.dataCache.resize(newCfg.cacheSizeBytes) {
+			b.sendNewPickerLocked()
```

Comment:
I think we need to think about this a little more. The fact that a configuration update is leading to multiple picker updates is not ideal. It is a little tricky here because of the order of locks.

Reply:
Hi Easwar, based on our discussions offline I have updated the PR and ensured that we are updating picker only once.